### PR TITLE
добавить делегирование

### PR DESCRIPTION
**Pull request by ya-ne-kit**
Добавлено делегирование согласования: 
предыдущее согласование закрывается со статусом DELEGATED, создается новый DocProcess с recipient'ом, которому делегировали согласование

### CODE REVIEW

**Comment by FokiDoki** [comment]
Jenkins, run CI

**Comment by FokiDoki** [comment]
Can one of the admins verify this patch?

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build #14: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/14/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/14/

**Comment by FokiDoki** [comment]
Jenkins, run CI

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build #18: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/18/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/18/

**Comment by FokiDoki** [comment]
Jenkins, run CI

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build `#53`: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/53/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/53/

**Review by vchumachenko — DISMISSED** [review]

**Comment by ya-ne-kit** [comment]
Jenkins, run CI

**Comment by ya-ne-kit** [comment]
Jenkins, run CI

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build `#89`: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/89/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/89/

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build `#99`: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/99/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/99/

File: src/main/java/ru/rosatom/documentflow/controllers/DocProcessController.java [hunk 1]
```diff
@@ -152,5 +153,14 @@ public DocProcessDto sendToCorrection( ProcessUpdateRequestDto processUpdateRequ
                 DocProcessDto.class);
     }
 
-
+    @PatchMapping("/processes/{processId}/delegate-to-other-user")
+    @PreAuthorize("@documentProcessSecurityService.isRecipient(#processUpdateRequestDto.processId, authentication.principal.id) && hasAuthority('USER')")
```
Comment (FokiDoki) [inline src/main/java/ru/rosatom/documentflow/controllers/DocProcessController.java:157]: Нужно добавить аннотации swagger-а

**Review by FokiDoki — CHANGES_REQUESTED** [review]

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build `#105`: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/105/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/105/

File: src/main/java/ru/rosatom/documentflow/controllers/DocProcessController.java (hunk 1 above)
Comment (FokiDoki) [reply src/main/java/ru/rosatom/documentflow/controllers/DocProcessController.java:159]: Плюс параметер recipientId передавать в  path а не в query т.к это не опциональный параметер

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build `#107`: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/107/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/107/

File: src/main/java/ru/rosatom/documentflow/controllers/DocProcessController.java (hunk 1 above)
Comment (ya-ne-kit) [reply src/main/java/ru/rosatom/documentflow/controllers/DocProcessController.java]: fixed

**Comment by FokiDoki** [comment]
Jenkins CI/CD: Build `#108`: Build success. Merge approved  

Details: https://jenkins.fokidoki.su:8888/job/docflow/108/
Blue ocean: https://jenkins.fokidoki.su:8888/blue/organizations/jenkins/docflow/detail/docflow/108/

**Review by FokiDoki — APPROVED** [review]

**Pull request merged**